app_server/app/policies/Authenticator.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * Singleton used for authenticating users
+ * @module app/policies/Authenticator
+ */
+var rfr = require('rfr');
+var util = require('util');
+var Promise = require('bluebird');
+var bcrypt = Promise.promisifyAll(require('bcrypt'));
+
+var SocialMediaAdapter = rfr('app/adapters/social_media/SocialMediaAdapter');
+var Service = rfr('app/services/Service');
+var Utility = rfr('app/util/Utility');
+
+var logger = Utility.createLogger(__filename);
+
+function Authenticator() {
+}
+var Class = Authenticator.prototype;
+
+Class.ERRORS = {
+  RETRIEVE_PROFILE: 'Error retrieving user\'s social media profile',
+};
+
+/**
+ * Authenticate a user through a social media platform
+ * @param platformType the name of the social media platform
+ * @param credentials the user's credentials in that platform
+ * @return {Promise} of the user's profile and credentials for worldscope
+ *                   or null if failed to authenticate user
+ */
+Class.authenticateUser = function (platformType, credentials) {
+  logger.info('Authenticating with %s', platformType);
+
+  var profilePromise =
+    Promise.method(function getSocialMediaAdapter() {
+      return new SocialMediaAdapter(platformType, credentials);
+    })().then(function (adapter) { return adapter.getUser(); });
+
+  var userPromise = profilePromise.then(function receiveProfile(profile) {
+    if (!profile || profile instanceof Error || !('id' in profile)) {
+      throw new Error(Class.ERRORS.RETRIEVE_PROFILE + ' ' +
+                      JSON.stringify(profile));
+    }
+
+    return Service.getUserByPlatform(platformType, profile.id);
+  });
+
+  return Promise.all([profilePromise, userPromise]).bind(this)
+  .spread(function (profile, user) {
+    if (!user || user instanceof Error) {
```

Comment:
Here too.
Should there be an indentation here?
It seems to pass the linter, after all.
We should keep it consistent, though, I remember seeing another place where there is indentation in such a place (line after `return`).